aodntools/timeseries_products/aggregated_timeseries.py

```diff
@@ -319,231 +235,201 @@ def source_file_attributes(download_url_prefix, opendap_url_prefix):
-
-                    if 'PRES_REL' in varnames:
-                        DF['PRES_REL'] = nc.PRES_REL.squeeze()
-                        try:
-                            applied_offset.append(nc.PRES_REL.applied_offset)
-                        except:
-                            applied_offset.append(np.nan)
-                        if 'PRES_REL_quality_control' in varnames:
-                            DF['PRES_REL_quality_control'] = nc.PRES_REL_quality_control.squeeze()
-                        else:
-                            DF['PRES_REL_quality_control'] = np.repeat(0, nobs)
-                    else:
-                        DF['PRES_REL'] = np.repeat(FILLVALUE, nobs)
-                        DF['PRES_REL_quality_control'] = np.repeat(9, nobs)
-                        applied_offset.append(np.nan)
-
-
-                    ## select only in water data
-                    DF = DF[(DF['TIME']>=time_deployment_start) & (DF['TIME']<=time_deployment_end)]
-
-                    ## append data
-                    variableMainDF = pd.concat([variableMainDF, DF], ignore_index=True, sort=False)
-
-
-                    # append auxiliary data
-                    variableAuxDF = variableAuxDF.append({'source_file': file,
-                                                          'instrument_id': nc.attrs['deployment_code'] + '; ' + nc.attrs['instrument'] + '; ' + nc.attrs['instrument_serial_number'],
-                                                          'LONGITUDE': nc.LONGITUDE.squeeze().values,
-                                                          'LATITUDE': nc.LATITUDE.squeeze().values,
-                                                          'NOMINAL_DEPTH': get_nominal_depth(nc)}, ignore_index = True)
-                    fileIndex += 1
-                else:
-                    rejected_files.append(file)
-                    bad_files.update({file: file_problems})
-        except:
-            print("FILE NOT FOUND: " + file, file=sys.stderr)
-
-    print()
-
-
-    ## rename indices
-    variableAuxDF.index.rename('INSTRUMENT', inplace=True)
-    variableMainDF.index.rename('OBSERVATION', inplace=True)
-
-    ## get the list of variables
-    varlist = list(variableMainDF.columns) + list(variableAuxDF.columns)
-
-    ## set variable attributes
-    add_variable_attribute = {'PRES_REL': {'applied_offset_by_instrument': applied_offset}}
+        with xr.open_dataset(os.path.join(input_dir, file)) as nc:
+            ## clip to in water data only
+            nc = in_water(nc)
+
+            error_list = check_file(nc, site_code, var_to_agg)
+            if not error_list:
+                varlen_list.append(len(nc.TIME))
+            else:
+                bad_files.update({file: error_list})
+                rejected_files.append(file)
+
+
+    ## remove bad files form the list
+    for file in bad_files.keys():
+        files_to_agg.remove(file)
+
+
+    varlen_list = [0] + varlen_list
```

Comment:
I need 0 as the start for the first slice of the netCDF variables. It could be done in a different way, but it will require some modifications to the logic of the code.